### DISCUSSION TITLE
Authorization for inputs

### DISCUSSION THREAD

**Discussion by arashi-dev** [MichalLytek/type-graphql#1156] [Q&A] [answered]
Hello

First of all, How can I authorize inputs? I know it is not possible directly, but is there any trick?

If not, I thought about creating some custom decorators which stores data and inputType property name to the `Reflect Metadata`, and modify the input based on settings by Decorator in the decorator on top of method.
But I need to get context as it has session id and user info in it. How can I get it via Reflect Metadata? 
this is how I want to use it:

in resolver:
```ts
@MethodAuth()   <------- here I want to get context
findUser(@Arg() @ArgAuth() filter: FindUserInput) {

}
```

in inputType:
```ts
@InputAuth({roles: ["ADMIN"]})
@Field({nullable: true})
email?: string
```

**Comment by MichalLytek** [comment, marked as answer]
The better design of API than authorized input fields is dedicated authorized mutations with enhanced input type.

**Reply by arashi-dev** [reply]
Alright, thanks 🙏
